ccm_web/server/src/invitation/cirrus-invitation.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@nestjs/common'
+import { ConfigService } from '@nestjs/config'
+import { Config } from '../config'
+import FormData from 'form-data'
+import { randomUUID } from 'crypto'
+import axios from 'axios'
+import { InvitationAPIError } from './invitation.errors'
+import { CanvasUserLoginEmail } from '../canvas/canvas.interfaces'
+import { HttpMethod } from '../api/api.utils'
+import baseLogger from '../logger'
+
+const logger = baseLogger.child({ filePath: __filename })
+
+@Injectable()
+export class CirrusInvitationService {
+  url: string
+  entityID: string
+  sponsorName: string
+  key: string
+  secret: string
+
+  constructor (private readonly configService: ConfigService<Config, true>) {
+    const invitationConfig = configService.get('invitation', { infer: true })
+    this.url = invitationConfig.apiURL
+    this.entityID = invitationConfig.apiEntityID
+    this.sponsorName = invitationConfig.apiSponsorName
+    this.key = invitationConfig.apiKey
+    this.secret = invitationConfig.apiSecret
+  }
+
+  async sendInvitations (users: CanvasUserLoginEmail[]): Promise<string> {
+    const userEmails: string[] = users.map(user => user.email)
+    const emailAddressCSV = `emailAddress\n${userEmails.join('\n')}`
+
+    const data = new FormData()
+    data.append('cfile', emailAddressCSV, 'fake_file_name.csv')
+    data.append('spEntityId', this.entityID)
+    data.append('sponsorEppn', this.sponsorName)
+    data.append('clientRequestID', 'ccm-' + randomUUID())
+
+    try {
+      const response = await axios({
+        method: HttpMethod.Post,
+        url: this.url,
+        auth: {
+          username: this.key,
+          password: this.secret
+        },
+        headers: { ...data.getHeaders() },
+        data: data
+      })
+
+      logger.debug(response.data)
+      return response.data
```

Comment:
It returns JSON.  I was hoping the switch from Axios to `HttpModule` would lead to easier parsing and handling of the return data, but that's not working out.

Even without that change, I can probably define a TS type that better describes the data.